File: 프로그래머스/unrated/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    
    data.sort(key=lambda x: (x[col - 1], -x[0]))
    
    for i in range(row_begin, row_end + 1):
        dataSum = 0
        
        for num in data[i - 1]:
            dataSum += num % i
        
        answer ^= dataSum
        
    
    return answer

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java
import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        Arrays.sort(data, (a, b) -> {
            if (a[col - 1] == b[col - 1]) return b[0] - a[0];
            
            return a[col - 1] - b[col - 1];
        });
        
        Queue<Integer> mods = new LinkedList<>();
        
        for (int i = row_begin - 1; i <= row_end - 1; i++) {
            int tmp = 0;
            
            for (int n : data[i]) {
                tmp += n % (i + 1);
            }

            mods.offer(tmp);
        }
        
        int answer = mods.poll();
        
        while (!mods.isEmpty()) {
            answer ^= mods.poll();
        }
        
        return answer;
    }
}